### ISSUE TITLE
Parallelizing Neural ODEs with Functorch

### ISSUE THREAD

**Issue by pharringtonp19** [DiffEqML/torchdyn#87]
I tend to work with relatively small datasets, so I have found one nice advantage of working with Jax is that I can train an ensemble of networks on a single GPU via [Will Whitney's approach](https://github.com/willwhitney/jax-parallel).

I see that [functorch](https://github.com/facebookresearch/functorch/blob/main/examples/ensembling/parallel_train.py) has made it possible to now do this in pytorch.

Question: Would it be feasible to make use of [functorch's](https://github.com/facebookresearch/functorch/blob/main/examples/ensembling/parallel_train.py) implementation to train an ensemble of neuralodes on a single GPU?

**Comment by Zymrael** [comment]
It should definitely be possible. We were considering `functorch` as a dependency of `torchdyn` since `vmap` and other vectorized operations are needed to implement some variants of fancier methods, such a [FwS MSLs](https://arxiv.org/abs/2106.03885), so perhaps a simple guide on how to do this could be useful.
